litgpt/generate/base.py

```diff
@@ -194,6 +194,120 @@ def generate_fn(
         yield from tokens[yielded_idx:]
 
 
+# TODO: Make include_eos work.
+# TODO: Rewrite unbatched generate_fn to use batched_generate_fn.
+@torch.inference_mode()
+def batched_generate_fn(
+    model: GPT,
+    prompts: torch.Tensor,
+    max_returned_tokens: int,
+    *,
+    sample_args: Union[list[dict], dict],
+    stop_tokens: Tuple[List[int], ...] = (),
+    include_prompt: bool,
+    include_eos: bool,
+) -> Iterator[list[Union[torch.Tensor, None]]]:
+
+    if isinstance(sample_args, dict):
+        sample_args = [sample_args] * len(prompts)
+
+    if prompts.ndim == 1:
+        prompts = prompts.unsqueeze(0)
+
+    assert prompts.ndim == 2, "Prompts must be a 2D tensor."
+    batch_size = prompts.size(0)
+    max_prompt_size = prompts.size(1)
+    device = prompts.device
+
+    assert all(prompt.size(0) == max_prompt_size for prompt in prompts), "For now, prompts must have the same length."
```

Comment:
You specify and use that prompts is a tensor, then this does not make much sense.

Reply:
The context around this changed, and it appears that I forgot to go back and edit it.